### DISCUSSION TITLE
Elasticsearch Output Format Question

### DISCUSSION THREAD

**Discussion by zimage** [dmachard/DNS-collector#658] [General]
Elasticseach has defined what they call their Elastic Common Schema, or ECS, that they recommend developers use when they import data into an Elasticsearch database. They have an [ECS for DNS queries and answers](https://www.elastic.co/guide/en/ecs/current/ecs-dns.html). I was wondering if go-dnscollector puts the data into ECS format when exporting it to Elasticsearch. I browsed the code quickly but couldn't find an obvious answer.

**Comment by dmachard** [comment]
The JSON model of the DNS-collector is described [here](https://github.com/dmachard/go-dnscollector/blob/main/docs/dnsjson.md). It's a generic model with more metadata than the ECS model.

We can add the ECS format directly in the [Elastic Logger](https://github.com/dmachard/go-dnscollector/blob/main/loggers/elasticsearch.go) and a new export mode 

```
elasticsearch:
  mode: ecs
```

Can you open a feature request ? And Feel free to submit a pull request :)
